packages/flutter_tools/test/src/package_config.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+/// Writes a `.dart_tool/package_config.json` file at [directory].
+///
+/// If directory is not specified, it will be `globals.fs.currentDirectory`;
+///
+/// It will contain a package entry for [mainLibName] with `rootUri` at
+/// [directory].
+///
+/// [otherLibs] maps other package names to their `rootUri` relative to `directory`.
+///
+/// Returns the `File` Object representing the package config.
+File writePackageConfigFile({
+  Directory? directory,
+  String mainLibName = 'my_app',
```

Comment:
Ditto, I would make this required instead of assuming a default.

Reply:
Done